lib/semantics/check-do.cc

```diff
@@ -213,9 +235,9 @@ class DoConcurrentBodyEnforce {
     return false;
   }
 
-  void addName(const std::optional<parser::Name> &nm) {
-    if (nm.has_value()) {
-      names_.insert(nm.value().source);
+  void addName(NamePtr nm) {
+    if (nm) {
```

Comment:
Can `AddName()` be legitimately called with a null pointer?  If not, use `CHECK()` or `DEREF()` instead.

Reply:
```AddName()``` gets called with a null pointer for statements that don't have names.